lib/pbench/agent/results.py

```diff
@@ -102,100 +122,78 @@ def make_result_tb(self):
         with mdlog_name.open("w") as fp:
             mdlog.write(fp)
 
-        # FIXME: subprocess "tar" command
         tarball = self.target_dir / f"{pbench_run_name}.tar.xz"
         try:
             with tarfile.open(tarball, mode="x:xz") as tar:
                 for f in self.result_dir.rglob("*"):
                     tar.add(os.path.realpath(f))
         except tarfile.TarError:
             self.logger.error(
-                "tar ball creation failed for {}, skipping", self.result_dir
+                "Tar ball creation failed for {}, skipping", self.result_dir
             )
             try:
                 tarball.unlink()
-            except OSError as exc:
-                if exc.errno != errno.ENOENT:
-                    self.logger.error("error removing failed tar ball, {}", tarball)
-            sys.exit(1)
-
-        self.make_md5sum(tarball)
+            except Exception as exc:
+                if not isinstance(exc, OSError) or exc.errno != errno.ENOENT:
+                    raise RuntimeError(
+                        f"Error removing failed tarball, '{tarball}', {exc}"
+                    )
 
         # The contract with the caller is to just return the full path to the
-        # created tar ball.
+        # created tarball.
         return str(tarball)
 
-    def make_md5sum(self, tarball):
-        tarball_md5 = Path(f"{tarball}.md5")
-        try:
-            hash_md5 = md5sum(tarball)
-            tarball_md5.write_text(f"{hash_md5} {tarball.name}\n")
-        except Exception:
-            self.logger.error("md5sum failed for {}, skipping", tarball)
-            try:
-                tarball.unlink()
-            except OSError as exc:
-                if exc.errno != errno.ENOENT:
-                    self.logger.error("error removing failed tar ball, {}", tarball)
-            try:
-                tarball_md5.unlink()
-            except OSError as exc:
-                if exc.errno != errno.ENOENT:
-                    self.logger.error(
-                        "error removing failed tar ball MD5, {}", tarball_md5
-                    )
-            sys.exit(1)
-
 
 class CopyResultTb:
+    """CopyResultTb - Copies tarball from agent to server with the help of
```

Comment:
Or "Use the HTTP `PUT` operation to upload a tarball to the server?" Placing the PUT operation in the role of active helper ("with the help of") sounds weird.

Reply:
Fixed.